File: invert_binary_tree.c
#include<stdio.h>
#include<stdlib.h>

/**
 ** Definition for a binary tree node.
 ** struct TreeNode {
 **     int val;
 **     struct TreeNode *left;
 **     struct TreeNode *right;
 ** };
 **/

struct TreeNode {
     int val;
     struct TreeNode *left;
     struct TreeNode *right;
};

struct TreeNode* invertTree(struct TreeNode* root) {
    struct TreeNode* current = root;
    struct TreeNode* temp;
    if(current){
        temp = current->left;
        current->left = current->right;
        current->right = temp;
        invertTree(current->left);
        invertTree(current->right);
    }
    return root;
}

File: py/tree/InvertBinaryTree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def invertTree(self, root):
        if root != None:
            invertedLeftTree = self.invertTree(root.left)
            invertedRightTree = self.invertTree(root.right)
            root.right = invertedLeftTree
            root.left = invertedRightTree   
        return root
